src/contract/types.ts

```diff
@@ -84,6 +84,7 @@ export type ClientOptions = {
       networkPassphrase?: string;
       accountToSign?: string;
     },
+    signer?: Keypair,
```

Comment:
We don't want to add a `signer` as an option to `signTransaction`. We want to use _the existing signature_, which comes from Freighter / [SEP-43](https://github.com/stellar/stellar-protocol/blob/eb401f932258c827a5b4a2e14aea939affcd2b02/ecosystem/sep-0043.md#wallet-interface-format), OR we want to allow passing a keypair. Like this:

```ts
signTransaction?: Keypair | (
  tx: XDR_BASE64,
  opts?: {
    network?: string;
    networkPassphrase?: string;
    accountToSign?: string;
  },
) => Promise<XDR_BASE64>;